chain/transaction.go

```diff
@@ -174,6 +154,28 @@ type Transaction struct {
 	stateKeys state.Keys
 }
 
+// newTransaction creates a Transaction and initializes the private fields.
+func newTransaction(base *Base, actions Actions, auth Auth) (*Transaction, error) {
+	tx := Transaction{
+		TransactionData: TransactionData{
+			Base:    base,
+			Actions: actions,
+		},
+		Auth: auth,
+	}
+	p := codec.NewWriter(0, consts.NetworkSizeLimit)
```

Comment:
Could we add a call to `UnsignedBytes` before calling `tx.Marshal(p)` to make sure that the `unsignedBytes` field is initialized as before?

This is correct as is since it will be lazily populated currently, but I'd prefer to keep the old behavior of making sure all private fields are populated when creating a new instance of a transaction.